src/unistd.rs

```diff
@@ -2136,3 +2146,424 @@ mod setres {
+    fn query(q: GroupQuery) -> Option<Result<Self>> {
+        let mut cbuf = match q {
+            GroupQuery::GidWithBufsize(_, size) | 
+            GroupQuery::NameWithBufsize(_, size) => {
+                vec![0 as c_char; size]
+            },
+            _ => {
+                vec![0 as c_char; PWGRP_BUFSIZE]
+            }
+        };
+
+        let mut grp: libc::group =  unsafe { mem::zeroed() };
+        let mut res = ptr::null_mut();
+
+        let i = {
+            unsafe { Errno::clear(); }
+
+            match q {
+                GroupQuery::GidWithBufsize(Gid(gid), _) |
+                GroupQuery::Gid(Gid(gid)) => { 
+                    unsafe { 
+                        libc::getgrgid_r(gid, &mut grp,
+                                         cbuf.as_mut_ptr(), 
+                                         cbuf.len(), &mut res) 
+                    }
+                },
+                GroupQuery::NameWithBufsize(name, _) |
+                GroupQuery::Name(name) => {
+                    unsafe {
+                        libc::getgrnam_r(CString::new(name).unwrap().as_ptr(), 
+                                         &mut grp, cbuf.as_mut_ptr(), cbuf.len(), &mut res) 
+                    }
+                },
+            }
+
+        };
+
+        match i {
+            0 => {
+                if !res.is_null() {
+                    Some(Ok(Group::from(res))) 
+                } else {
+                    None
+                }
+            },
+            _ => { Some(Err(Error::Sys(Errno::last()))) },
+        }
+    }
+}
+
+#[cfg(not(any(target_os = "android", 
+              target_os = "ios", 
+              target_os = "macos", 
+              target_env = "musl",
+              target_arch = "mips",
+              target_arch = "mips64",
+              target_arch = "mips64el",
+              target_arch = "mipsel",
+              target_arch = "s390x")))] 
+mod usergroupiter {
+    use libc;
+    use libc::{c_char};
+    use Result;
+    use errno::Errno;
+    use super::{Error, User, Group, PWGRP_BUFSIZE};
+    use std::{mem, ptr};
+
+    /// Used to get all of the users on the system.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use nix::unistd::Users;
+    /// eprintln!("Users on this system:\nname\tuid");
+    /// Users::new()
+    ///     .map(|e|e.map(
+    ///         |pw| println!("{}\t{}",
+    ///                       pw.name,
+    ///                       pw.uid)))
+    ///     .collect::<Vec<_>>();
+    /// 
+    /// ```
+    ///
+    /// # Safety
+    ///
+    /// This iterator should not be used in different threads without synchronization; while doing so
+    /// will not cause undefined behavior, because modern systems lack re-entrant versions of
+    /// `setpwent` and `endpwent`, it is very likely that iterators running in different threads will
+    /// yield different numbers of items.
+    #[derive(Debug)]
+    pub struct Users(pub usize);
```

Comment:
Is there any reason the user of this API needs to set the internal value? If not, you should make it private and add a getter to get the value instead. This is more common with Rust types and used elsewhere in `nix`.

Reply:
done.